datalad/interface/diff.py

```diff
@@ -0,0 +1,276 @@
+            no dataset is given, an attempt is made to identify the dataset
+            based on the input and/or the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        path=Parameter(
+            args=("path",),
+            metavar="PATH",
+            doc="""path to be evaluated""",
+            nargs="*",
+            constraints=EnsureStr() | EnsureNone()),
+        revision=Parameter(
+            args=('--revision',),
+            metavar='REVISION EXPRESSION',
+            nargs='?',
+            doc="""comparison reference specification. Three modes are
+            supported: 1) <revision> changes you have in your working tree
+            relative to the named revision (this can also be a branch name,
+            tag, commit or any label Git can understand). 2) <revision>..<revision>
+            changes between two arbitrary revisions. 3) <revision>...<revision>
+            changes on the branch containing and up to the second <revision>,
+            starting at a common ancestor of both revisions."""),
+        staged=Parameter(
+            args=("--staged",),
+            action="store_true",
+            doc="""get the changes already staged for a commit relative
+            to an optionally given revision (by default the most recent one)"""),
+        ignore_subdatasets=Parameter(
+            args=('--ignore-subdatasets',),
+            constraints=EnsureChoice('none', 'untracked', 'dirty', 'all'),
+            doc="""speed up execution by (partially) not evaluating the state of
+            subdatasets in a parent dataset. With "none" a subdataset is
+            considered modified when it either contains untracked or modified
+            content or its last saved state differs from that recorded in the
+            parent dataset. When "untracked" is used subdatasets are not
+            considered modified when they only contain untracked content (but
+            they are still scanned for modified content). Using "dirty" ignores
+            all changes to the work tree of subdatasets, only changes to the
+            revisions stored in the parent dataset are shown. Using "all" hides
+            all changes to subdatasets. Note, even with "all" recursive
+            execution will still report other changes in any existing
+            subdataset, only the subdataset record in a parent dataset
+            is not  evaluated."""),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit)
+
+    @staticmethod
+    @datasetmethod(name='diff')
+    @eval_results
+    def __call__(
+            path=None,
+            dataset=None,
+            revision='HEAD',
+            staged=False,
+            ignore_subdatasets='none',
+            recursive=False,
+            recursion_limit=None):
+        if not dataset and not path:
+            # act on the whole dataset if nothing else was specified
+            dataset = curdir
```

Comment:
Now I remember, `require_dataset` create a `Dataset` instance (not necessarily needed), and it also walks the path upward. This would lead to path annotations above any given request or the PWD -- not desired.